File: Effective Approach/effective_approach.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

void Effective_Approach()
{  ll array_size;
   cin>>array_size;
   ll array[array_size];
   map<ll,vector<ll>> position;
   for(int i = 0;i < array_size;i++)
   {   cin>>array[i];
       vector<ll> v;
       v.push_back(i+1); 
       v.push_back(array_size-i); 
       position[array[i]] = v;
   }
   int total_queries;
   ll total_search_from_start = 0,total_search_from_end = 0;
   cin>>total_queries;
   ll queries[total_queries];
   for(int i = 0;i < total_queries;i++)
   {   cin>>queries[i];
       total_search_from_start += position[queries[i]][0];
       total_search_from_end += position[queries[i]][1];
       
   }
   cout<<total_search_from_start<<" "<<total_search_from_end<<"\n";
   
       
    
}


int main() {
	int remained_test_cases = 1 ;
// 	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   Effective_Approach();
	    remained_test_cases--;
	}
	return 0;
}

File: Effective Approach/effective_approach.py
def Effective_Approach():
    total_elements = int(input())
    array = list(map(int,input().split()))
    total_queries = int(input())
    position = {}
    
    for num in range(total_elements):
        position[array[num]] = [num+1,total_elements-num]
        
    total_check_from_start = 0
    total_check_from_end = 0
    
    queries = list(map(int,input().split()))
    for q in range(total_queries):
        total_check_from_start += position[queries[q]][0]
        total_check_from_end += position[queries[q]][1]
        
    return [total_check_from_start,total_check_from_end]    
        
    
remained_test_cases = 1 
# remained_test_cases = int(input())
while remained_test_cases > 0:
    print(*Effective_Approach())
    remained_test_cases -= 1 